### PR TITLE
Add login and signup functionality

### PR DESCRIPTION
**Pull request by aroerick**
User signup and authentication with graphcool.

### CODE REVIEW

File: js/components/AccountInput/AccountInput.js [hunk 1]
```diff
@@ -2,12 +2,15 @@ import React from 'react';
...
   return (
     <TextInput
-      style={styles.input}
+      autoCapitalize="none"
+      style={
+        invalid ? [styles.input, styles.invalid] : [styles.input, styles.valid]
+      }
       placeholder={placeholder}
-      placeholderTextColor="#0082B580"
+      placeholderTextColor={invalid ? '#d84934' : '#0082B580'}
```
Comment (siddharthparmar7) [inline js/components/AccountInput/AccountInput.js:13]: hash values are supposed to be 6 chars only. Please change it.

File: js/navigation/NavigationLayout.js [hunk 2]
```diff
@@ -9,7 +9,22 @@ import CreateItem from './../screens/CreateItem';
 import Listings from './../screens/Listings';
 import Messages from './../screens/Messages';
 import Profile from './../screens/Profile';
+import AccountForm from '../screens/AccountForm';
 
+export const accountFormStack = createStackNavigator(
```
Comment (siddharthparmar7) [inline js/navigation/NavigationLayout.js]: Let's call it authStack, more descriptive.

File: js/navigation/NavigationLayout.js [hunk 3]
```diff
@@ -9,7 +9,22 @@ import CreateItem from './../screens/CreateItem';
...
+  {
+    accountForm: {
+      screen: AccountForm,
+    },
+  },
+  {
+    navigationOptions: {
+      headerStyle: {
+        height: 0,
```
Comment (siddharthparmar7) [inline js/navigation/NavigationLayout.js]: If you do not need header, use header: none

File: js/screens/AccountForm/AccountForm.js [hunk 4]
```diff
@@ -14,40 +15,89 @@ const AccountForm = ({ formState, toggleForm }) => {
...
+          onSubmit={
+            formState
+              ? values => {
+                  if (values.password !== values.verifyPassword) return;
+                  delete values.verifyPassword;
+                  signup.mutation({ variables: { ...values } })
+                  .then(() => navigation.navigate('Listings'));
+                }
+              : values => {
+                  delete values.verifyPassword && delete values.username;
```
Comment (siddharthparmar7) [inline js/screens/AccountForm/AccountForm.js]: Why are you deleting passwords?

File: js/screens/AccountForm/styles.js [hunk 5]
```diff
@@ -19,8 +19,9 @@ const styles = StyleSheet.create({
   },
```
Comment (siddharthparmar7) [inline js/screens/AccountForm/styles.js:18]: can we use dimensions and flex properties here for styling our logo?

File: js/screens/AccountForm/styles.js [hunk 6]
```diff
@@ -36,6 +37,17 @@ const styles = StyleSheet.create({
...
+    position: 'absolute',
+    fontFamily: fonts.main,
+    color: colors.red,
+  },
+  loginError: {
+    top: 90,
+  },
+  signupError: {
+    top: 193,
   }
```
Comment (siddharthparmar7) [inline js/screens/AccountForm/styles.js]: Can we use flex properties here? instead of top?

**Review by siddharthparmar7 — CHANGES_REQUESTED** [review]
Please add an activity indicator or a loading screen to show that login is in process.

Please add a log out button for testing.

Nice work using react-adpot. If you are going to follow this pattern, please have a separate file for all of your quries and import them as required. 

Nice Work.

File: js/components/AccountInput/AccountInput.js (hunk 1 above)
Comment (aroerick) [reply js/components/AccountInput/AccountInput.js:13]: The design specifies a 50% transparency on the main brand blue and react-native docs said rgba hex were valid for transparency. Is there a different preferred way to do this?

**Review by siddharthparmar7 — APPROVED** [review]
Thank you! nice work!

**Pull request merged**